app/code/Magento/Webapi/Controller/Rest/RequestProcessorPool.php

```
@@ -36,9 +36,22 @@ public function __construct($requestProcessors = [])
      */
     public function getProcessor(\Magento\Framework\Webapi\Rest\Request $request)
     {
-        foreach ($this->requestProcessors as $processor) {
-            if ($processor->canProcess($request)) {
-                return $processor;
+        foreach ($this->requestProcessors as $processorData) {
+
+            /**
+             * Condition was created to keep backward compatibility
+             */
+            if (is_object($processorData)) {
+                if ($processorData->canProcess($request)) {
+                    return $processorData;
+                }
+            } else {
+                if (isset($processorData['processor']) && isset($processorData['validator'])) {
```

To avoid hard-detecting bugs with invalid config (e.g. typo in keys) we need to validate array structure in a constructor and throw `InvalidArgumentException` if inappropriate data passed (not silently ignore them)